src/AppBundle/EventListener/RedirectToPreferredLocaleListener.php

```diff
@@ -81,6 +81,11 @@ public function onKernelRequest(GetResponseEvent $event)
         if (!$event->isMasterRequest() || '/' !== $request->getPathInfo()) {
             return;
         }
+        // Ignore requests from referrers with the same HTTP host in order to prevent
+        // changing language for users who possibly already selected it for this application.
+        if (0 === stripos($request->server->get('HTTP_REFERER'), $request->getSchemeAndHttpHost())) {
```

Comment:
It's the same things I think. They both are equal and I also can directly modify both too. But there're no any security issues for this PR, only usage convenience. I prefer to keep SERVER's `HTTP_REFERER` because it well documented in [PHP docs](http://php.net/manual/en/reserved.variables.server.php) (to understand what is the `$request->headers->get('REFERER')` here - more difficult for newcomers). What do you think?